### DISCUSSION TITLE
Unused Queues Deletion

### DISCUSSION THREAD

**Discussion by Pramod34** [rabbitmq/rabbitmq-server#11127] [Questions] [answered]
Hi Team,

We have a 100's hosts sending messages to our RabbitMQ cluster with 3 nodes, we have multiple consumers consuming those messages and reply/acknowledge to hosts. Some of our hosts logout when they are idle for sometime, our users will reconnect hosts by login which will create a new queue and old queue is left and unused in the RabbitMQ.

Can you please best suggest us a way to clean up the unused queues in RabbitMQ which should be on-going process which keeps on cleaning unused queues?

**Note:** 
1. We have a use case where our hosts cannot close/delete queues when logout. 
2. We also have a use case where queues with messages should not get deleted.

It's really helpful please let us know if any suggestions or solutions

Thanks,
Pramod

**Comment by lukebakken** [comment, marked as answer]
Thanks for using RabbitMQ.

If you haven't yet noticed, RabbitMQ's documentation usually can answer what you are looking for -

https://www.rabbitmq.com/docs/queues#temporary-queues
https://www.rabbitmq.com/docs/ttl#queue-ttl

* If you use auto-delete queues, you could lose messages.
* If you use queue TTL, you could lose messages if they are not consumed prior to queue TTL expiration.

In your case, you will have to write your own process to enumerate the queues in your system (via the HTTP API), and delete those that meet your deletion criteria.
